### PR TITLE
WIP: Add MySQL metrics collector

### PR DESCRIPTION
**Pull request by susana-garcia**
## Summary

* Add MySQL metrics collector.

## Checklist

- [X] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [X] Update tests.

### CODE REVIEW

File: pkg/service/dbaas/service.go [hunk 1]
```diff
@@ -89,6 +99,10 @@ func (s *Service) fetchManagedDBaaS(ctx *Context) error {
 	for _, groupVersionResource := range groupVersionResources {
 		managedResources, err := s.k8sClient.Resource(groupVersionResource).List(ctx, metav1.ListOptions{})
 		if err != nil {
+			// fixme
+			if err.Error() == "the server could not find the requested resource" {
+				continue
+			}
```
Comment (glrf) [inline pkg/service/dbaas/service.go:105]: What exactly should be fixed here?

If it's the matching on error message then I agree. AFAICT you should be able to just use `errors.IsNotFound` from `apimachinery` here or am I missing something?

File: pkg/database/postgres.go [hunk 2]
```diff
@@ -13,8 +13,6 @@ type productDBaaS struct {
 }
 
 var (
-	// Hobbyist1 plan
-	Hobbyist1 = productDBaaS{Plan: "hobbyist-1", Target: "1411", Amount: 0.06148}
```
Comment (glrf) [inline pkg/database/postgres.go:17]: Nitpick: This should technically be in it's own PR, but I don't mind

File: pkg/database/types.go [hunk 3]
```diff
@@ -25,14 +29,17 @@ const (
...
+	defaultUnitDBaaS     = "Instances"
 )
 
 // exoscale service types to query billing Database types
 var (
 	billingTypes = map[string]string{
-		"pg": queryDBaaSPostgres,
+		"pg":    queryDBaaSPostgres,
+		"mysql": queryDBaaSMysql,
 	}
```
Comment (glrf) [inline pkg/database/types.go:43]: Opensearch is missing here

**Review by glrf — COMMENTED** [review]
Could you either update the title to include opensearch or (even better) split this PR into two. One for mysql and one for opensearch?

File: pkg/service/dbaas/service.go (hunk 1 above)
Comment (ZuGao) [reply pkg/service/dbaas/service.go:105]: exactly what you just proposed. THanks for the tip :)

**Comment by ZuGao** [comment]
I will close this PR in favour of [PR](https://github.com/vshn/exoscale-metrics-collector/pull/28) as we decided to add them all in one go.

File: pkg/service/dbaas/service.go (hunk 1 above)
Comment (susana-garcia) [reply pkg/service/dbaas/service.go:105]: @glrf I didn't have time to finish it yesterday 😅

Comment (glrf) [reply pkg/service/dbaas/service.go:105]: I realized that afterwards.  Sorry I thought this was done when I reviewed it :see_no_evil:

Comment (susana-garcia) [reply pkg/service/dbaas/service.go:105]: hehe, no problem, I forgot to add the `wip` tag until later too 😄
